AdminUI/src/app/services/tier-service/tier.service.ts

```diff
@@ -32,15 +32,32 @@ export class TierService {
     }
 
     getTierById(id: string): Observable<HttpResponseEnvelope<Tier>> {
-        // TODO: Missing an API endpoint to get a tier by ID
-        const httpOptions = {
-            params: new HttpParams().set('id', id),
-        };
+        // Missing an API endpoint to get a tier by ID
+        const maxPageSize = 200;
 
-        return this.http.get<HttpResponseEnvelope<Tier>>(
-            this.apiUrl,
-            httpOptions
-        );
+        return new Observable<HttpResponseEnvelope<Tier>>((subscriber) => {
+            this.getTiers(0, maxPageSize).subscribe({
+                next: (data: PagedHttpResponseEnvelope<Tier>) => {
+                    const tier = data.result.find((tier) => {
+                        return tier.id == id;
+                    });
```

Comment:
Can be simplified:
```suggestion
                    const tier = data.result.find(t => t.id == id);
```
I will approve this PR anyway. You can make this change with your next PR.